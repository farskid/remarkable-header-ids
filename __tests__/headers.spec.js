// @ts-nocheck

const Remarkable = require("remarkable");
const HeaderIdsPlugin = require("../index");
const content = require("../mockData");

function parseAllHeaders(
  content,
  levels = [1, 2, 3, 4, 5, 6]
) {
  return new Remarkable.Remarkable()
    .use(
      HeaderIdsPlugin({
        levels,
        anchorClassName: "this-is-anchor-in-header",
        anchorText: "#",
        headerId: (slug) =>
          `header-that-has-this-anchor-${slug}`,
      })
    )
    .render(content);
}

describe("headers", () => {
  test("Renders anchor tags for all headers successfully", () => {
    expect(parseAllHeaders(content)).toMatchSnapshot();
  });

  test("Renders only H1 anchors", () => {
    expect(parseAllHeaders(content, [1])).toMatchSnapshot();
  });

  test("Renders only H2 anchors", () => {
    expect(parseAllHeaders(content, [2])).toMatchSnapshot();
  });

  test("Renders only H3 anchors", () => {
    expect(parseAllHeaders(content, [3])).toMatchSnapshot();
  });

  test("Renders only H4 anchors", () => {
    expect(parseAllHeaders(content, [4])).toMatchSnapshot();
  });

  test("Renders only H5 anchors", () => {
    expect(parseAllHeaders(content, [5])).toMatchSnapshot();
  });

  test("Renders only H6 anchors", () => {
    expect(parseAllHeaders(content, [6])).toMatchSnapshot();
  });

  test("Renders combination of levels", () => {
    expect(
      parseAllHeaders(content, [1, 4, 6])
    ).toMatchSnapshot();
  });
});
