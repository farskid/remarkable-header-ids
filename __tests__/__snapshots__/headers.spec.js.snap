// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headers Renders anchor tags for all headers successfully 1`] = `
"<h1 id=\\"header-that-has-this-anchor-h1\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h1\\" href=\\"#h1\\">#</a>H1</h1>
<p>Some text for h1</p>
<h2 id=\\"header-that-has-this-anchor-h2\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h2\\" href=\\"#h2\\">#</a>H2</h2>
<p>Some text for h2</p>
<h3 id=\\"header-that-has-this-anchor-h3\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h3\\" href=\\"#h3\\">#</a>H3</h3>
<p>Some text for h3</p>
<h4 id=\\"header-that-has-this-anchor-h4\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h4\\" href=\\"#h4\\">#</a>H4</h4>
<p>Some text for h4</p>
<h5 id=\\"header-that-has-this-anchor-h5\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h5\\" href=\\"#h5\\">#</a>H5</h5>
<p>Some text for h5</p>
<h6 id=\\"header-that-has-this-anchor-h6\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h6\\" href=\\"#h6\\">#</a>H6</h6>
<p>Some text for h6</p>
"
`;

exports[`headers Renders combination of levels 1`] = `
"<h1 id=\\"header-that-has-this-anchor-h1\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h1\\" href=\\"#h1\\">#</a>H1</h1>
<p>Some text for h1</p>
<h2>H2</h2>
<p>Some text for h2</p>
<h3>H3</h3>
<p>Some text for h3</p>
<h4 id=\\"header-that-has-this-anchor-h4\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h4\\" href=\\"#h4\\">#</a>H4</h4>
<p>Some text for h4</p>
<h5>H5</h5>
<p>Some text for h5</p>
<h6 id=\\"header-that-has-this-anchor-h6\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h6\\" href=\\"#h6\\">#</a>H6</h6>
<p>Some text for h6</p>
"
`;

exports[`headers Renders only H1 anchors 1`] = `
"<h1 id=\\"header-that-has-this-anchor-h1\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h1\\" href=\\"#h1\\">#</a>H1</h1>
<p>Some text for h1</p>
<h2>H2</h2>
<p>Some text for h2</p>
<h3>H3</h3>
<p>Some text for h3</p>
<h4>H4</h4>
<p>Some text for h4</p>
<h5>H5</h5>
<p>Some text for h5</p>
<h6>H6</h6>
<p>Some text for h6</p>
"
`;

exports[`headers Renders only H2 anchors 1`] = `
"<h1>H1</h1>
<p>Some text for h1</p>
<h2 id=\\"header-that-has-this-anchor-h2\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h2\\" href=\\"#h2\\">#</a>H2</h2>
<p>Some text for h2</p>
<h3>H3</h3>
<p>Some text for h3</p>
<h4>H4</h4>
<p>Some text for h4</p>
<h5>H5</h5>
<p>Some text for h5</p>
<h6>H6</h6>
<p>Some text for h6</p>
"
`;

exports[`headers Renders only H3 anchors 1`] = `
"<h1>H1</h1>
<p>Some text for h1</p>
<h2>H2</h2>
<p>Some text for h2</p>
<h3 id=\\"header-that-has-this-anchor-h3\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h3\\" href=\\"#h3\\">#</a>H3</h3>
<p>Some text for h3</p>
<h4>H4</h4>
<p>Some text for h4</p>
<h5>H5</h5>
<p>Some text for h5</p>
<h6>H6</h6>
<p>Some text for h6</p>
"
`;

exports[`headers Renders only H4 anchors 1`] = `
"<h1>H1</h1>
<p>Some text for h1</p>
<h2>H2</h2>
<p>Some text for h2</p>
<h3>H3</h3>
<p>Some text for h3</p>
<h4 id=\\"header-that-has-this-anchor-h4\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h4\\" href=\\"#h4\\">#</a>H4</h4>
<p>Some text for h4</p>
<h5>H5</h5>
<p>Some text for h5</p>
<h6>H6</h6>
<p>Some text for h6</p>
"
`;

exports[`headers Renders only H5 anchors 1`] = `
"<h1>H1</h1>
<p>Some text for h1</p>
<h2>H2</h2>
<p>Some text for h2</p>
<h3>H3</h3>
<p>Some text for h3</p>
<h4>H4</h4>
<p>Some text for h4</p>
<h5 id=\\"header-that-has-this-anchor-h5\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h5\\" href=\\"#h5\\">#</a>H5</h5>
<p>Some text for h5</p>
<h6>H6</h6>
<p>Some text for h6</p>
"
`;

exports[`headers Renders only H6 anchors 1`] = `
"<h1>H1</h1>
<p>Some text for h1</p>
<h2>H2</h2>
<p>Some text for h2</p>
<h3>H3</h3>
<p>Some text for h3</p>
<h4>H4</h4>
<p>Some text for h4</p>
<h5>H5</h5>
<p>Some text for h5</p>
<h6 id=\\"header-that-has-this-anchor-h6\\"><a class=\\"this-is-anchor-in-header\\" id=\\"h6\\" href=\\"#h6\\">#</a>H6</h6>
<p>Some text for h6</p>
"
`;
